governance/pyth_staking_sdk/src/pyth-staking-client.ts

```diff
@@ -0,0 +1,410 @@
+import { AnchorProvider, BN, Program } from "@coral-xyz/anchor";
+import { Connection, PublicKey } from "@solana/web3.js";
+import { type Staking } from "../types/staking";
+import * as StakingIdl from "../idl/staking.json";
+import * as IntegrityPoolIdl from "../idl/integrity_pool.json";
+import * as PublisherCapsIdl from "../idl/publisher_caps.json";
+import {
+  getConfigAddress,
+  getPoolConfigAddress,
+  getStakeAccountCustodyAddress,
+  getStakeAccountMetadataAddress,
+} from "./pdas";
+import type { GlobalConfig, PoolConfig, PoolDataAccount } from "./types";
+import {
+  type StakeAccountPositions,
+  StakeAccountPositionsAnchor,
+} from "./staking/accounts";
+import type { IntegrityPool } from "../types/integrity_pool";
+import {
+  type Account,
+  createTransferInstruction,
+  getAccount,
+  getAssociatedTokenAddress,
+} from "@solana/spl-token";
+import {
+  sendTransactions,
+  TransactionBuilder,
+} from "@pythnetwork/solana-utils";
+import { convertBigIntToBN, convertBNToBigInt } from "./utils/bn";
+import type { AnchorWallet } from "@solana/wallet-adapter-react";
+import type { PublisherCaps } from "../types/publisher_caps";
+import { getUnlockSchedule } from "./utils/vesting";
+
+export type PythStakingClientConfig = {
+  connection: Connection;
+  wallet: AnchorWallet;
+};
+
+export class PythStakingClient {
+  connection: Connection;
+  wallet: AnchorWallet;
+  provider: AnchorProvider;
+  stakingProgram: Program<Staking>;
+  integrityPoolProgram: Program<IntegrityPool>;
+  publisherCapsProgram: Program<PublisherCaps>;
+
+  constructor(config: PythStakingClientConfig) {
+    this.connection = config.connection;
+    this.wallet = config.wallet;
+    this.provider = new AnchorProvider(this.connection, this.wallet, {
+      skipPreflight: true,
+    });
+    this.stakingProgram = new Program(StakingIdl as Staking, this.provider);
+    this.integrityPoolProgram = new Program(
+      IntegrityPoolIdl as IntegrityPool,
+      this.provider
+    );
+    this.publisherCapsProgram = new Program(
+      PublisherCapsIdl as PublisherCaps,
+      this.provider
+    );
+  }
+
+  async setGlobalConfig(config: GlobalConfig) {
```

Comment:
why setGlobalConfig instead of initGlobalConfig. This is not really a setter tbh since it only works once anyway

Reply:
fixed.